Get subdomain multisite tests working properly

jazzsequence opened the pull request:
We're not running the full setup-playwright-tests script because subdomain multisites don't need most of the setup. However, it does need _some_ of the setup (specifically copying the changes from the PR to the fixture site), so we need to manually add those back into that workflow.

In the future we might be able to run just portions of the script for the purpose of subdomain multisite so we're not duplicating code here.

~This PR is blocked by #147 -- the changes the robot fixed were part of what was updated in that PR. Merging this PR before that one will create a commit that gets dropped during deploy.~

This change includes some tweaks to the actual playwright tests and some deviations in the setup. It was discovered in running the tests that the subdomain fixture site was getting reset (GraphQL plugin being deleted, multisite config resetting) -- those have been added to the setup -- but in troubleshooting those it was hypothesized that an incorrect GraphQL endpoint url _could have been_ the issue (it wasn't, multisite not being set up was the issue), or that the environment variables weren't being processed correctly (that wasn't it either, but that was only discovered after breaking the Playwright tests into separate tests -- one for the main site and one for the subdomain). An additional passed environment variable is now supported, however, for GraphQL endpoint, which means that we can eventually remove the handling in the test for different endpoint urls for _subdirectory_ multisite (not included in this PR as it doesn't relate to the subdomain testing).

jazzsequence committed:
add copying the changes in the PR to the subdomain tests

pwtyler commented on .github/workflows/playwright.yml:
```diff
@@ -385,6 +386,37 @@ jobs:
+            exit 1
+          fi
+          # Check SUBDOMAIN_INSTALL value
+          SUBDOMAIN_INSTALL=$(terminus wp "${{ env.SITE_ID }}".dev -- config get SUBDOMAIN_INSTALL)
+          if [[ "${SUBDOMAIN_INSTALL}" != "1" ]]; then
+            # SUBDOMAIN_INSTALL should be true.
+              echo -e "${RED}Subdomain configuration not found!${RESET}"
+              exit 1
+          fi
```
```suggestion
          if ! terminus wp "${{ env.SITE_ID }}".dev -- config is-true SUBDOMAIN_INSTALL; then
              echo -e "Subdomain configuration not found!"
              exit 1
          fi
```

pwtyler commented on .github/workflows/playwright.yml:
```diff
@@ -385,6 +386,37 @@ jobs:
+
+          echo "Checking WordPress install status"
+          terminus wp "${{ env.SITE_ID }}".dev -- cli info
+          if ! terminus wp "${{ env.SITE_ID }}".dev -- config is-true MULTISITE; then
+            echo -e "${RED}Multisite not found!${RESET}"
+            exit 1
+          fi
```
```suggestion
          if ! terminus wp "${{ env.SITE_ID }}".dev -- config is-true MULTISITE; then
            echo -e "Multisite not found!"
            exit 1
          fi
```

jazzsequence committed:
switch to git mode before we try pushing things

jazzsequence committed:
don't save in a variable

Co-authored-by: Phil Tyler <philip@tylerdigital.com>

jazzsequence committed:
remove color tags

jazzsequence committed:
remove color tags

jazzsequence committed:
exclude node_modules from rsync
playwright runs on the ci, not on the fixture

jazzsequence committed:
exclude node_modules from rsync
playwright runs on the ci, not on the fixture

jazzsequence committed:
add multisite config to the fixture site
this shouldn't need to be done every time, so we check if multisite is true and add the config if it's not true

jazzsequence committed:
fix quoting

jazzsequence committed:
fix quoting

jazzsequence committed:
remove lingering ${type} vars and stray quote

jazzsequence committed:
remove lingering ${type} vars and stray quote

jazzsequence committed:
don't run playwright tests if files we don't care about were affected

jazzsequence committed:
don't run playwright tests if files we don't care about were affected

jazzsequence committed:
move the wait to earlier
we can't check the config if it hasn't finished being updated

jazzsequence committed:
give up on the multisite checking
we should know it already is multisite so we'll just check that the domains exist

jazzsequence committed:
install graphql as part of the workflow

jazzsequence committed:
clear ache after installing graphql

jazzsequence committed:
break the subdomain sites up
so we can pass different values in

jazzsequence committed:
allow the graphql endpoint to be passed
we can use this to fix the subdirectory test, too, but later

jazzsequence committed:
copy the application.php for subdom
this is getting reset, too, for some reason

jazzsequence commented:
@pwtyler re-requesting review since I've made changes since your ✅

pwtyler commented on .github/workflows/playwright.yml:
```diff
@@ -385,13 +401,63 @@ jobs:
+          echo "Checking site URLs"
+          SITE_URL_TEST=$(curl -s -o /dev/null -w "%{http_code}" ${{ env.SITE_URL }})
+          SUBDOMAIN_URL_TEST=$(curl -s -o /dev/null -w "%{http_code}" ${{ env.SUBDOMAIN_URL }})
+          if [ $SITE_URL_TEST -ne 200 ] || [ $SUBDOMAIN_URL_TEST -ne 200 ]; then
+            echo "One or more site URLs are not returning a 200 status code. Exiting."
+            exit 1
+          fi
```
might be useful to debug print what each are
```suggestion
          if [ $SITE_URL_TEST -ne 200 ] || [ $SUBDOMAIN_URL_TEST -ne 200 ]; then
            echo "One or more site URLs are not returning a 200 status code. Exiting."
            echo "${{ env.SITE_URL }} - ${SITE_URL_TEST}"
            echo "${{ env.SUBDOMAIN_URL }} - ${SUBDOMAIN_URL_TEST}"
            exit 1
          fi
```

jazzsequence committed:
add more debugging info to failure output

Co-authored-by: Phil Tyler <philip@tylerdigital.com>
